### PR TITLE
Overhaul pt2: Writing your own application, tooling and guidelines

### PR DESCRIPTION
**Pull request by SergioGasquez**
- Updates Writing your own application and tooling chapters
- Adds style guidelines

### CODE REVIEW

File: rust-doc-style-guide.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+  > 🚨 **Warning**: Use in critical circumstances only, e.g., for irreversible actions or actions potentially harmful to hardware, software, etc.
+
+In markdown:
+
+```md
+> ⚠️ **Note**: Write your note.
+```
+
+## Appenndix A Existing Style Guides
```
Comment (f-hollow) [inline rust-doc-style-guide.md:307]:
```suggestion
## Appendix A Existing Style Guides
```
My typo.

File: rust-doc-style-guide.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
+<!-- omit in toc -->
+# Rust Documentation Style Guide
```
Comment (f-hollow) [inline rust-doc-style-guide.md:2]: Do you think we can rename it to "Rust Doc Style Guide?"

It is much shorter and still intuitive enough.

File: src/tooling/debugging/vscode-debugging.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ There is also a possibility to debug with graphical output directly in Visual St
 
 ### Hardware Setup
 
-ESP32 doesn't have a built-in JTAG interface so you have to connect an external JTAG adapter to the ESP32 board, for example, [ESP-Prog](https://docs.espressif.com/projects/espressif-esp-iot-solution/en/latest/hw-reference/ESP-Prog_guide.html) can be used.
+ESP32 doesn't have a built-in JTAG interface so, you have to connect an external JTAG adapter to the ESP32 board, for example, [ESP-Prog](https://docs.espressif.com/projects/espressif-esp-iot-solution/en/latest/hw-reference/ESP-Prog_guide.html) can be used.
```
Comment (f-hollow) [inline src/tooling/debugging/vscode-debugging.md:9]:
```suggestion
ESP32 doesn't have a built-in JTAG interface, so you have to connect an external JTAG adapter to the ESP32 board, for example, [ESP-Prog](https://docs.espressif.com/projects/espressif-esp-iot-solution/en/latest/hw-reference/ESP-Prog_guide.html) can be used.
```

File: src/tooling/debugging/vscode-debugging.md [hunk 4]
```diff
@@ -61,13 +61,13 @@ ESP32 doesn't have a built-in JTAG interface so you have to connect an external
...
+ESP32-C3 with **revision 3** **does** have a built-in JTAG interface, and you don't have to connect an external device to be able to debug. To get the chip revision, run the `cargo espflash board-info` command.
 
 ### Hardware Setup
 
 If your ESP32-C3's revision is lesser than 3, follow these instructions, if you have revision 3 you can jump to the [**Set up VSCode**](#set-up-vscode-1) step.
 
-ESP32-C3 **revision 1** and **revision 2** don't have a built-in JTAG interface so you have to connect an external JTAG adapter to the ESP32-C3 board, for example, [ESP-Prog](https://docs.espressif.com/projects/espressif-esp-iot-solution/en/latest/hw-reference/ESP-Prog_guide.html) can be used.
+ESP32-C3 **revision 1** and **revision 2** don't have a built-in JTAG interface so, you have to connect an external JTAG adapter to the ESP32-C3 board, for example, [ESP-Prog](https://docs.espressif.com/projects/espressif-esp-iot-solution/en/latest/hw-reference/ESP-Prog_guide.html) can be used.
```
Comment (f-hollow) [inline src/tooling/debugging/vscode-debugging.md:70]:
```suggestion
ESP32-C3 **revision 1** and **revision 2** don't have a built-in JTAG interface, so you have to connect an external JTAG adapter to the ESP32-C3 board, for example, [ESP-Prog](https://docs.espressif.com/projects/espressif-esp-iot-solution/en/latest/hw-reference/ESP-Prog_guide.html) can be used.
```

Just a quick reference. This is a compound sentence with a coordinating conjunction (for, and, so, etc.). The comma should be placed before the coordinating conjunction. For details, see [this article](https://study.com/skill/learn/using-commas-with-compound-and-complex-sentences-explanation.html)

File: src/writing-your-own-application/generate-project/index.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Using Dev Containers in the templates
+
+Both template repositories have a prompt for Dev Containers support, when using Dev Containers in the templates it will add support for:
+-  [VS Code Dev Containers]
+-  [GitHub Codespaces]
+
+Dev Containers use the [`idf-rust` container image], that was explained in the [Using Container section] of the installation chapter, and provide an environment ready to develop Rust applications for Espressif chips with no installation required. Dev Containers also have integration with [Wokwi simulator], to simulate the project, and allow flashing from the container using [web flash].
+
+For more details about on Dev Containers, see [Dev Container] section of the template README.
```
Comment (f-hollow) [inline src/writing-your-own-application/generate-project/index.md:39]:
```suggestion
For more details on Dev Containers, see [Dev Container] section of the template README.
```

**Review by f-hollow — COMMENTED** [review]
@SergioGasquez 

A couple of nitpicks.

Once the changes are merged, I will read through the chapters more thoroughly to see if there the content structure can/should be improved, etc.

LGTM!

File: rust-doc-style-guide.md (hunk 2 above)
Comment (SergioGasquez) [reply rust-doc-style-guide.md:2]: Sounds good to me, I dont really mind. 

Also, we could probably include the style guide from Rust book somewhere? https://github.com/rust-lang/book/blob/main/style-guide.md

Comment (jessebraham) [reply rust-doc-style-guide.md]: I think using the full word should always be preferable in documentation, personally.

Comment (SergioGasquez) [reply rust-doc-style-guide.md:2]: Let's use the full name, so there is no room for confusion. Change is now reverted!

Comment (f-hollow) [reply rust-doc-style-guide.md:2]: The reason is the general trend on the internet to shorten anything possible to make it "scannable." These days we have increasingly more abbreviations and abbreviations made of other abbreviations, such as AIoT.

But I agree that using full words is better, so let's stick with "documentation."

Comment (f-hollow) [reply rust-doc-style-guide.md:2]: @SergioGasquez 

> Also, we could probably include the style guide from Rust book somewhere? https://github.com/rust-lang/book/blob/main/style-guide.md

The style guide from Rust book is already included in the appendix under the name [book-ja](https://github.com/rust-lang-ja/book-ja/blob/master-ja/style-guide.md).

Please replace that link with the one you found.

Comment (SergioGasquez) [reply rust-doc-style-guide.md:2]: I've updated the links from book-ja (Japanese version of the book) to the English one and removed the one I included.

**Pull request merged**